sdk/nodejs/runtime/mocks.ts

```diff
@@ -65,7 +64,21 @@ export class MockMonitor {
 
     public async invoke(req: any, callback: (err: any, innerResponse: any) => void) {
         try {
-            const result = this.mocks.call(req.getTok(), deserializeProperties(req.getArgs()), req.getProvider());
+            const tok = req.getTok();
+            const inputs = deserializeProperties(req.getArgs());
+
+            if (tok === "pulumi:pulumi:getResource") {
```

Comment:
Presumably we are OK handling this on behalf of the user, so they don't have to implement it themselves.

If there ever was a scenario where it's interesting to allow mocking calls for `"pulumi:pulumi:getResource"` in the future, I could imagine updating the `Mocks` interface to support an optional `getResource` function, that we could call if it was set instead of using the implementation here.

Reply:
We should be implementing this, as it is a builtin that deals only in state that is internal to the engine.